### PR TITLE
Fix damage from exploding beds being cancelled

### LINKED ISSUE #3872: Beds don't do player damage when tnt-explosion is disabled
### Information

**Full output of `/ess version`:**

```
>.... [13:33:25 INFO]: Server version: 1.16.4-R0.1-SNAPSHOT git-Paper-345 (MC: 1.16.4)
>.... [13:33:25 INFO]: EssentialsX version: 2.18.2.0
>.... [13:33:25 INFO]: LuckPerms version: 5.2.35
>.... [13:33:25 INFO]: PlaceholderAPI version: 2.10.10-DEV-111
>.... [13:33:25 INFO]: Vault version: 1.7.3-b131
>.... [13:33:25 INFO]: EssentialsXProtect version: 2.18.2.0
>.... [13:33:25 INFO]: EssentialsXSpawn version: 2.18.2.0
>.... [13:33:25 INFO]: Citizens version: 2.0.27-SNAPSHOT (build 2190)
>.... [13:33:25 INFO]: Towny version: 0.96.5.7
```

**Server startup log:**

https://gist.github.com/DatSpace/97f6545c94b7503f285d6528c8936f32

**EssentialsX config:**

https://gist.github.com/DatSpace/1cc29b1526e38e0991e9b7c86ea5ed75

### Details

**Description:**  

Beds in nether do not give player damage, even though they break blocks etc.

**Steps to reproduce:**  

Simply set tnt-explosion: true in the config. You can have everything else to false, it doesnt matter. Then go to nether in survival mode and try to sleep. Everything will break, but player takes 0 damage.

**Expected behavior:**  

I expected the player to take explosive damage

**Console stack trace:**  

```

```

**Screenshots:**

### PR DESCRIPTION
**Pull request by ploppyperson**
This PR fixes #3872 

Proposed fix: Add check if exploding block material is tnt.

Environments tested: git-Paper-325 (MC: 1.16.4)

### CODE REVIEW

**Comment by pop4959** [comment]
What happens with beds if tnt explosion damage is disabled? You should make sure that servers are still able to disable beds.

**Comment by ploppyperson** [comment]
> What happens with beds if tnt explosion damage is disabled? You should make sure that servers are still able to disable beds.

If protect.prevent.tnt-playerdamage is enabled, then TNT doesn't do damage but beds still do.

**Comment by pop4959** [comment]
> If protect.prevent.tnt-playerdamage is enabled, then TNT doesn't do damage but beds still do.

Sorry if not clear, what I'm really asking is with this PR are servers able to both:
- enable tnt damage but disable bed damage
- enable bed damage but disable tnt damage

Just want to make sure because this was catching both before, but it should be possible to control them individually. In addition, you should check that any other exploding blocks are handled correctly too.

**Comment by ploppyperson** [comment]
> > If protect.prevent.tnt-playerdamage is enabled, then TNT doesn't do damage but beds still do.
> 
> Sorry if not clear, what I'm really asking is with this PR are servers able to both:
> 
>     * enable tnt damage but disable bed damage
> 
>     * enable bed damage but disable tnt damage
> 
> 
> Just want to make sure because this was catching both before, but it should be possible to control them individually. In addition, you should check that any other exploding blocks are handled correctly too.

The PR just means that the config option for tnt damage only disables tnt damage and not bed damage. I can add config options for bed damage if required?

**Comment by pop4959** [comment]
Would only be required if the change breaks behavior that previously was expected.

File: EssentialsProtect/src/main/java/com/earth2me/essentials/protect/EssentialsProtectEntityListener.java [hunk 1]
```diff
@@ -69,7 +70,8 @@ public void onEntityDamage(final EntityDamageEvent event) {
                 event.setCancelled(true);
                 return;
             }
-            if (prot.getSettingBool(ProtectConfig.prevent_tnt_explosion) && cause == DamageCause.BLOCK_EXPLOSION && !(target instanceof Player && shouldBeDamaged(user, "tnt"))) {
+            final EntityDamageByBlockEvent edbEvent = (EntityDamageByBlockEvent) event;
```
Comment (JRoy) [inline EssentialsProtect/src/main/java/com/earth2me/essentials/protect/EssentialsProtectEntityListener.java:73]: dangerous cast, need to be checked

**Review by JRoy — CHANGES_REQUESTED** [review]

File: EssentialsProtect/src/main/java/com/earth2me/essentials/protect/EssentialsProtectEntityListener.java (hunk 1 above)
Comment (ploppyperson) [reply EssentialsProtect/src/main/java/com/earth2me/essentials/protect/EssentialsProtectEntityListener.java:73]: It's checked on line 64

**Review by JRoy — APPROVED** [review]

**Pull request merged**